### DISCUSSION TITLE
Question about LDP and notes encoding

### DISCUSSION THREAD

**Discussion by denisgottardello** [lenmus/lomse#396] [Q&A]
Please have a look at the picture
![Screenshot_20240619_142701](https://github.com/lenmus/lomse/assets/43038562/31969c38-c626-4b8d-81ae-be1b44c3dfd0)

The relative lpd code is the following:

`(score (vers 2.0)(instrument (staves 2) (musicData (clef G p1) (clef F p2) (key G)

(n f4 q p1)
(n +f4 q p1)
(n -g4 q p1)

(barline end))))
`

Can someone explain to me:
The first note (f4) means +f4, key 66 of the piano.
The second note is not possible. Is true?
The third note is -g4 (correctly), key 66 of the piano.
Is the first note correct or should be +f4?

**Comment by cecilios** [comment]
In LDP notes accidentals are writen as you would do in a writen score. Thus
first note, f4, will be displayed as F but will be played as F# due to key
signature.
Second note, +f4, will be also a F# but displaying a cautionary accidental.
Last note, -g4, will be a G flat and is enarmonic to F#

**Comment by cecilios** [comment]
First and last notes are correct. Second is also correct but will force to
display a cautionary accidental (no parenthesis, they are not yet possible
in LDP)

**Comment by denisgottardello** [comment]
Thank you for the explanation.
I have completed the implementation of reading exercise in my software for musicians using your library.
The musician reads and play using a midi keyboard. If the playd note is correct the curson move over.
http://www.denisgottardello.it/PianoJokeTouch/index.php

**Comment by cecilios** [comment]
Glad to know. Feel free to ask if you have more questions
